### PR TITLE
Fix for shellcheck

### PR DESCRIPTION
**Pull request by rickwu666666**
Result of "./manage.py test"
https://pastebin.canonical.com/p/Kby5dxGPHd/

### CODE REVIEW

**Review by patliuu — APPROVED** [review]
LGTM

File: checkbox-provider-ce-oem/units/eeprom/test-plan.pxu [hunk 1]
```diff
@@ -20,7 +20,8 @@ unit: test plan
 _name: EEPROM auto tests
 _description: Automated EEPROM tests for devices
 include:
-    ce-oem-eeprom/read-write
+nested_part:
+    com.canonical.certification::eeprom-automated
```
Comment (stanley31huang) [inline checkbox-provider-ce-oem/units/eeprom/test-plan.pxu:25]: I think we could just remove this test plan and put the `com.canonical.certification::eeprom-automated` test plan into ce-oem-automated directly

File: checkbox-provider-ce-oem/units/eeprom/test-plan.pxu [hunk 2]
```diff
@@ -33,4 +34,5 @@ unit: test plan
 _name: EEPROM auto tests
 _description: Automated EEPROM tests for devices
 include:
-    after-suspend-ce-oem-eeprom/read-write
+nested_part:
+    com.canonical.certification::after-suspend-eeprom-automated
```
Comment (stanley31huang) [inline checkbox-provider-ce-oem/units/eeprom/test-plan.pxu:38]: same here

File: checkbox-provider-ce-oem/units/rtc/test-plan.pxu [hunk 3]
```diff
@@ -37,8 +35,6 @@ unit: test plan
...
 _description: Automated after suspend multiple RTC tests
 bootstrap_include:
-    ce-oem-rtc_list
 include:
-    after-suspend-ce-oem-rtc/rtc_number
-    after-suspend-ce-oem-rtc/rtc_alarm_.*
-    after-suspend-ce-oem-rtc/rtc_clock_.*
\ No newline at end of file
+nested_part:
+    com.canonical.certification::after-suspend-rtc-automated
```
Comment (stanley31huang) [inline checkbox-provider-ce-oem/units/rtc/test-plan.pxu:40]: same here

**Comment by hanhsuan** [comment]
There are some remains warning in the code

`ce-oem-eeprom/read-write`
```sh
In - line 2:
if [ $? != 0 ]
     ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.


In - line 13:
  if [ $? == 0 ]
       ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.


In - line 18:
    if [ $? == 0 ]
         ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
```

`ce-oem-rtc/rtc_number`

```sh
In - line 1:
if [ -z $TOTAL_RTC_NUM ]; then 
        ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
if [ -z "$TOTAL_RTC_NUM" ]; then 


In - line 4:
SYSFS_RTC_NUM=`ls /dev/rtc[0-9] | wc -l`
              ^------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.
               ^--------------^ SC2012: Use find instead of ls to better handle non-alphanumeric filenames.

Did you mean: 
SYSFS_RTC_NUM=$(ls /dev/rtc[0-9] | wc -l)


In - line 7:
if [ $SYSFS_RTC_NUM != $TOTAL_RTC_NUM ]; then
     ^------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
if [ "$SYSFS_RTC_NUM" != $TOTAL_RTC_NUM ]; then
```

`ce-oem-rtc_list`

```sh

In - line 1:
rtcs=$(ls /dev/rtc[0-9] | awk -F "/" '{print $3}')
       ^--------------^ SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
```

`ce-oem-rtc/rtc_clock_{rtc}`

```sh
In - line 2:
rtc_date=$(date -d @$rtctime)
                    ^------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
rtc_date=$(date -d @"$rtctime")


In - line 4:
systime_date=$(date -d @$systime)
                        ^------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
systime_date=$(date -d @"$systime")


In - line 5:
result=$(($systime - $rtctime))
          ^------^ SC2004: $/${} is unnecessary on arithmetic variables.
                     ^------^ SC2004: $/${} is unnecessary on arithmetic variables.


In - line 8:
    echo $rtc_date
         ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
    echo "$rtc_date"


In - line 11:
    echo "System clock=" $systime_date
                         ^-----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
    echo "System clock=" "$systime_date"


In - line 12:
    echo "RTC clock=" $rtc_date
                      ^-------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
    echo "RTC clock=" "$rtc_date"
```

**Comment by rickwu666666** [comment]
> There are some remains warning in the code
> 
> `ce-oem-eeprom/read-write`
> 
> ```shell
> In - line 2:
> if [ $? != 0 ]
>      ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
> 
> 
> In - line 13:
>   if [ $? == 0 ]
>        ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
> 
> 
> In - line 18:
>     if [ $? == 0 ]
>          ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
> ```
> 
> `ce-oem-rtc/rtc_number`
> 
> ```shell
> In - line 1:
> if [ -z $TOTAL_RTC_NUM ]; then 
>         ^------------^ SC2086: Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
> if [ -z "$TOTAL_RTC_NUM" ]; then 
> 
> 
> In - line 4:
> SYSFS_RTC_NUM=`ls /dev/rtc[0-9] | wc -l`
>               ^------------------------^ SC2006: Use $(...) notation instead of legacy backticked `...`.
>                ^--------------^ SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
> 
> Did you mean: 
> SYSFS_RTC_NUM=$(ls /dev/rtc[0-9] | wc -l)
> 
> 
> In - line 7:
> if [ $SYSFS_RTC_NUM != $TOTAL_RTC_NUM ]; then
>      ^------------^ SC2086: Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
> if [ "$SYSFS_RTC_NUM" != $TOTAL_RTC_NUM ]; then
> ```
> 
> `ce-oem-rtc_list`
> 
> ```shell
> In - line 1:
> rtcs=$(ls /dev/rtc[0-9] | awk -F "/" '{print $3}')
>        ^--------------^ SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
> ```
> 
> `ce-oem-rtc/rtc_clock_{rtc}`
> 
> ```shell
> In - line 2:
> rtc_date=$(date -d @$rtctime)
>                     ^------^ SC2086: Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
> rtc_date=$(date -d @"$rtctime")
> 
> 
> In - line 4:
> systime_date=$(date -d @$systime)
>                         ^------^ SC2086: Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
> systime_date=$(date -d @"$systime")
> 
> 
> In - line 5:
> result=$(($systime - $rtctime))
>           ^------^ SC2004: $/${} is unnecessary on arithmetic variables.
>                      ^------^ SC2004: $/${} is unnecessary on arithmetic variables.
> 
> 
> In - line 8:
>     echo $rtc_date
>          ^-------^ SC2086: Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
>     echo "$rtc_date"
> 
> 
> In - line 11:
>     echo "System clock=" $systime_date
>                          ^-----------^ SC2086: Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
>     echo "System clock=" "$systime_date"
> 
> 
> In - line 12:
>     echo "RTC clock=" $rtc_date
>                       ^-------^ SC2086: Double quote to prevent globbing and word splitting.
> 
> Did you mean: 
>     echo "RTC clock=" "$rtc_date"
> ```

That's weird. Those jobs related to EEPROM and RTC should be removed from this PR.
https://github.com/canonical/checkbox-provider-ce-oem/pull/29/commits/238c4e18f8177c459d9ca9cf08273d6f53e452a0
https://github.com/canonical/checkbox-provider-ce-oem/pull/29/commits/789632f8ac2abaed1059f831fb9e127f773da6d4

**Comment by hanhsuan** [comment]
> That's weird. Those jobs related to EEPROM and RTC should be removed from this PR. [238c4e1](https://github.com/canonical/checkbox-provider-ce-oem/commit/238c4e18f8177c459d9ca9cf08273d6f53e452a0) [789632f](https://github.com/canonical/checkbox-provider-ce-oem/commit/789632f8ac2abaed1059f831fb9e127f773da6d4)

Oh, it might be I copy your code from your branch to mine, and the removed code in your branch is existed in mine branch.

**Review by hanhsuan — APPROVED** [review]
LGTM

**Review by stanley31huang — APPROVED** [review]
LGTM

**Pull request merged**